apps/web/src/components/Twitter/TwitterSetting.tsx

```diff
@@ -0,0 +1,174 @@
+import { TwitterSettingMutation } from '~/generated/TwitterSettingMutation.graphql';
+import { usePromisifiedMutation } from '~/hooks/usePromisifiedMutation';
+import TwitterIcon from '~/icons/Twittericon';
+
+import { Button } from '../core/Button/Button';
+import colors from '../core/colors';
+import InteractiveLink from '../core/InteractiveLink/InteractiveLink';
+import { HStack, VStack } from '../core/Spacer/Stack';
+import { BaseM } from '../core/Text/Text';
+import Toggle from '../core/Toggle/Toggle';
+
+type Props = {
+  queryRef: TwitterSettingFragment$key;
+};
+
+export default function TwitterSetting({ queryRef }: Props) {
+  const query = useFragment(
+    graphql`
+      fragment TwitterSettingFragment on Query {
+        viewer {
+          ... on Viewer {
+            socialAccounts {
+              twitter {
+                username
+                display
+              }
+            }
+          }
+        }
+      }
+    `,
+    queryRef
+  );
+
+  const [disconnectTwitter] = usePromisifiedMutation<TwitterSettingDisconnectMutation>(graphql`
+    mutation TwitterSettingDisconnectMutation($input: SocialAccountType!) {
+      disconnectSocialAccount(accountType: $input) {
+        __typename
+
+        ... on DisconnectSocialAccountPayload {
+          viewer {
+            ... on Viewer {
+              socialAccounts {
+                twitter {
+                  username
+                  display
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  `);
+
+  const [updateTwitterDisplay] = usePromisifiedMutation<TwitterSettingMutation>(graphql`
+    mutation TwitterSettingMutation($input: UpdateSocialAccountDisplayedInput!) {
+      updateSocialAccountDisplayed(input: $input) {
+        __typename
+
+        ... on UpdateSocialAccountDisplayedPayload {
+          viewer {
+            ... on Viewer {
+              socialAccounts {
+                twitter {
+                  username
+                  display
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  `);
+
+  const handleDisconnectTwitter = useCallback(() => {
+    disconnectTwitter({
+      variables: {
+        input: 'Twitter',
+      },
+    });
+  }, [disconnectTwitter]);
+
+  const handleUpdateTwitterDisplay = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const displayed = event.target.checked;
+      updateTwitterDisplay({
+        variables: {
+          input: {
+            displayed,
+            type: 'Twitter',
+          },
+        },
+
+        optimisticUpdater: (store) => {
+          const viewer = store.getRoot().getLinkedRecord('viewer');
+          const socialAccounts = viewer?.getLinkedRecord('socialAccounts');
+          const twitter = socialAccounts?.getLinkedRecord('twitter');
+          twitter?.setValue(displayed, 'display');
+        },
```

Comment:
we should be able to use an optimistic response here instead. I generally lean towards those since they're typesafe / graphql safe.

If we change how the mutation works, it won't be obvious that this code doesn't work anymore

Reply:
One of the reason, I used `optimisticUpdater` because I want it to have an instant toggle


Example when I used the `optimisticResponse`. There is a slightly delay on the toggle
https://user-images.githubusercontent.com/4480258/221331871-9a8aa20e-6aa3-4d04-9341-b89af912e83d.mp4

Reply:
```jsx
      updateTwitterDisplay({
        variables: {
          input: {
            displayed,
            type: 'Twitter',
          },
        },
        optimisticResponse: {
          updateSocialAccountDisplayed: {
            __typename: 'UpdateSocialAccountDisplayedPayload',
            viewer: {
              __typename: 'Viewer',
              socialAccounts: {
                __typename: 'SocialAccounts',
                twitter: {
                  __typename: 'Twitter',
                  username: query.viewer?.socialAccounts?.twitter?.username,
                  display: displayed,
                },
              },
            },
          },
        },
      });
```

Reply:
This is because your viewer doesn't have an `id` field in the response. Relay doesn't know which `Viewer` to update

Reply:
Got it. Now it works! Thanks terence 🙌🏻